### ISSUE TITLE
why does this code is required?

### ISSUE THREAD

**Issue by zxyyxzz** [MegviiDetection/video_analyst#206]
````
if task in ["track", "vos"]:
         # build dummy dataset for purpose of dataset setup (e.g. caching path list)
         logger.info("Build dummy AdaptorDataset")
         dummy_py_dataset = AdaptorDataset(
            task,
            cfg,
            num_epochs=cfg.num_epochs,
            nr_image_per_epoch=cfg.nr_image_per_epoch,
            seed=seed,
        )
        logger.info("Read dummy training sample")
        dummy_sample = dummy_py_dataset[0]  # read dummy sample
        del dummy_py_dataset, dummy_sample




Hi, i read your code, but i can't understand the code here, Isn't it redundant here?

Hope you can reply me,  thank you!~~ :-)

**Comment by zxyyxzz** [comment]
@lzx1413

**Comment by MARMOTatZJU** [comment]
@zxyyxzz  Please paste the link to starting line here so that I can trace to the context of this block easily.

From this stage, I can only guess that some static class members will be initialized when the first object is instantiated, and this is required before the instantiation of the first real dataset. So a dummy object is built here just for initialization. 

Specifically, the following comment line in the code explains the using purpose:
`build dummy dataset for purpose of dataset setup (e.g. caching path list)`

**Comment by MARMOTatZJU** [comment]
Closed as no further discussion. Feel free to reopen it again.
